Automattic/wp-calypso
# Social previews: add Mastodon preview

Related to #77179

## Proposed Changes

This PR adds Mastodon to the list of previews in the `social-previews` package. It also instantiates the preview in the sharing preview at `/posts/:site`.

In more detail:
- it adds shared post helpers to `client/blocks/sharing-preview-pane/utils.js`
- it conditionally adds the Mastodon preview in `SharingPreviewPane`.
- it implemented the base preview for Mastodon in the `social-previews` package.

## Testing Instructions



### Prerequisites
- Make sure you have a Mastodon account
- Note the blog id of your test site
- Add the proper blog sticker to your site, as described in D111014-code
- Visit `wordpress.com/marketing/connections/:site` and connect your Mastodon account

### Testing
- Spin up Calypso by running this branch locally or by using the live link below
- Visit `/posts/:site`
- Click _Share_ in the post options
- Then click _Preview_
- Notice you can see the Mastodon preview
- Test with various combinations of content and custom message/image

![Screenshot 2023-05-24 at 4 20 58 PM](https://github.com/Automattic/wp-calypso/assets/1620183/3947616f-d623-403a-b161-de4419a900dc)


## Pre-merge Checklist
